compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -54,11 +55,22 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
   private def elimRepeated(tp: Type)(using Context): Type = tp.stripTypeVar match
     case tp @ MethodTpe(paramNames, paramTypes, resultType) =>
       val resultType1 = elimRepeated(resultType)
-      val paramTypes1 =
-        if paramTypes.nonEmpty && paramTypes.last.isRepeatedParam then
-          val last = paramTypes.last.translateFromRepeated(toArray = tp.isJavaMethod)
-          paramTypes.init :+ last
-        else paramTypes
+      val paramTypes1 = paramTypes match
+        case init :+ last if last.isRepeatedParam =>
```

Comment:
This refactoring has the performance consequence that the `init` will always be extracted even if there is no repeat param at the end. Previously, constructing the `init` list would be delayed until we knew there was a repeat param.

Are you OK with that?

Reply:
Yeah, I though it was a bit clearer that way but that's a lot of extra allocations indeed, fixed.